src/Storages/ObjectStorage/DataLakes/IDataLakeMetadata.h

```diff
@@ -58,6 +58,8 @@ class IDataLakeMetadata : boost::noncopyable
     virtual std::optional<size_t> totalRows(ContextPtr) const { return {}; }
     virtual std::optional<size_t> totalBytes(ContextPtr) const { return {}; }
 
+    virtual std::shared_ptr<void> getOpaque() const { return nullptr; }
```

Comment:
It would be good to have at least a comment about the purpose of this thing in the interface. TBH, is it really necessary to be so abstract? Maybe just put here what you needed in the first place, and return nothing if there is no such thing?